editors/sc-ide/core/sc_process.cpp

```diff
@@ -53,6 +53,8 @@ ScProcess::ScProcess(Settings::Manager* settings, QObject* parent):
     connect(mIpcServer, SIGNAL(newConnection()), this, SLOT(onNewIpcConnection()));
     connect(this, SIGNAL(stateChanged(QProcess::ProcessState)), this,
             SLOT(onProcessStateChanged(QProcess::ProcessState)));
+
+    connect(settings, &Settings::Manager::requestRestartLanguage, this, &ScProcess::restartLanguage);
```

Comment:
Wouldn't it be "cleaner" to have the connection within the dialog?

The sc_process shouldn't need to be aware of any settings dialog, but the settings dialog of the lang could be aware of an process b/c it also displays the respective warning.

Reply:
Yeah, but how to I expose the process in that part of the code? I couldn't figure it out.

Reply:
Ah, that's much better! Changed, thanks.